pkg/controllers/common/constants.go

```diff
@@ -127,6 +127,11 @@ const (
 	PodUnschedulableThresholdAnnotation = InternalPrefix + "pod-unschedulable-threshold"
 	// AutoMigrationInfoAnnotation contains auto migration information.
 	AutoMigrationInfoAnnotation = DefaultPrefix + "auto-migration-info"
+	// ObservedAnnotationKeysAnnotation contains the observed annotation keys should be synced from source object
+	// when reconcile federated objects.
+	ObservedAnnotationKeysAnnotation = DefaultPrefix + "observed-annotations"
```

Comment:
```suggestion
    FederatePrefix = "federate." + DefaultPrefix
    
	// ObservedAnnotationKeysAnnotation contains annotation keys observed in the last reconcile. It will be in the format of `a,b|c,d`, where `a` and `b` are the keys that are synced from source annotations to federated object annotations.
	ObservedAnnotationKeysAnnotation = FederatePrefix + "observed-annotations"
```